File: 0216.组合总和III/0216-组合总和III.java
class Solution {
    public List<List<Integer>> combinationSum3(int k, int n) {
        List<List<Integer>> result=new ArrayList();
        List<Integer> path=new ArrayList();
        backtrack(k,n,1,result,path);
        return result;
    }
    private void backtrack(int k,int n,int cur,List<List<Integer>> result,List<Integer> path){
        if (k==0 && n==0){
            result.add(new ArrayList(path));
            return;
        }
        if (k<0 || n<0) return;
        for (int i=cur;i<10;i++){
            path.add(i);
            backtrack(k-1,n-i,i+1,result,path);
            path.remove(path.size()-1);
        }
        return;
    }
}

File: 0216.组合总和III/0216-组合总和III.py
class Solution(object):
    def combinationSum3(self, k, n):
        """
        :type k: int
        :type n: int
        :rtype: List[List[int]]
        """

        def backtrack(k,n,cur,result,path):
            if (k==0 and n==0):
                result.append(path[:])
                return
            if n<0 or k<0:
                return
            for i in range(cur,10):
                path.append(i)
                backtrack(k-1,n-i,i+1,result,path)
                path.pop()
            return
        result=[]
        backtrack(k,n,1,result,[])
        return result